src/js/media-chrome-button.js

```diff
@@ -1,7 +1,9 @@
 import MediaChromeHTMLElement from './media-chrome-html-element.js';
 import { defineCustomElement } from './utils/defineCustomElement.js';
+import { isServer } from './utils/browser-env.js';
+import { createTemplate } from './utils/createTemplate.js';
 
-const template = document.createElement('template');
+const template = createTemplate();
```

Comment:
Wondering if we should instead add `createElement` as a mock function on your `Document` shim. My main driver of that is trying to avoid abstracting enough things into functions that this becomes yet another framework rather than "just using HTML/JS". The hole I see myself digging though is having to add a mock function for any window/document function used in the future...

Reply:
> My main driver of that is trying to avoid abstracting enough things into functions that this becomes yet another framework rather than "just using HTML/JS". The hole I see myself digging though is having to add a mock function for any window/document function used in the future...

Exactly my thoughts. I don't want to build out a whole advanced mock document & window.

The alternative would be updating each file that uses `createTemplate` to look more like this:

```
import { Document, isServer } from './utils/browser-env';

const template = isServer ? Document.createElement('template') : {};

template.innerHTML = ``;
```

It feels bit messy to have to throw in those ternary operators everywhere.

Options:

1. keep `utils/createElement`
2. add `Document.createElement` to our shim, but with the understanding that we're not going to build out a complex shim for every window/document function
3. remove `utils/createElement`, but add conditionals to check that we're not in a server whenever we want to `createElement('template')` (which we do in every file)

I want to avoid (3) but I don't feel too strongly between (1) and (2). Thoughts?

Reply:
Yeah, depends on what we're prioritizing most.
* DRY Code -> # 1 `utils/createTemplate`
* Be clear what's happening, help people create their own components -> # 3 add conditionals
* Maybe somewhere in between -> # 2 Document.createElement

I lean towards 3. Also because it makes it really easy to remove when server-side renderers get their act together and support web components.

I might call it something like `./utils/server-safe-globals`, `import { document } from './utils/server-safe-globals'; `

> Exactly my thoughts. I don't want to build out a whole advanced mock document & window.

At the same time, it's not any more work than creating new function wrappers for everything that uses a window/doc property.

Reply:
Let me know if you want to do this, or not, or if you want me to do it, and we'll get this pulled in.